scripts/release.sh

```diff
@@ -117,12 +117,32 @@ function check_tag_correct() {
 }
 
 # build_release builds the actual release binaries.
-#   arguments: <version-tag> <build-system(s)> <build-tags> <ldflags>
+#   arguments: <version-tag> <build-system(s)> <build-tags> <ldflags> <go-version>
 function build_release() {
   local tag=$1
   local sys=$2
   local buildtags=$3
   local ldflags=$4
+  local goversion=$5
+
+  # We will use the `GOTOOLCHAIN` environment variable to specify the Go version
+  # to use by the Go build command. Only go1.21 and newer support this
+  # environment variable.
+  #
+  # Now we will check to ensure that the installed Go version is 1.21 or newer.
+  min_required_installed_go_version="1.21"
+  installed_go_version=$(go version | awk '{print $3}' | sed 's/go//')
+
+  if [ "$(printf '%s\n' "$min_required_installed_go_version" "$installed_go_version" | sort -V | head -n1)" = "$min_required_installed_go_version" ]; then
```

Comment:
This line is very long. Perhaps we should extract the first part into a variable as well?